### ISSUE TITLE
Permission error for key files

### ISSUE THREAD

**Issue by WolfspiritM** [Homegear/Homegear-Docker#10]
Hi,

I have this image running on a synology nas and had troubles getting ssl to work.
The reason is that at least on my nas when mounting the config directory /etc/homegear to a folder of the nas, then the owner of the mounted files is always root:root.

This is no problem except for the ssl keys.

I've now added a chmod homegear:homegear for the homegear.key, homegear.crt and the dh1024.pem file to the start.sh script so when I restart the container everything works fine.
But when I create a new container these changes are gone of course.

Is it possible to include that chown into the start.sh even if the files already exist to make sure the homegar user can read the key?

**Comment by hfedcba** [comment]
Should be fixed in the next image. Thanks!

**Comment by WolfspiritM** [comment]
@hfedcba Thanks! I gave it a try and I think it should also be done for \*.pem files and also for ca/private/\*.key and \*.pem files as it still didn't come up.

Is there a reason why the whole folder is "chown"ed to root instead of the ${USER}?

**Comment by hfedcba** [comment]
Done.

> Is there a reason why the whole folder is "chown"ed to root instead of the ${USER}?

Mainly so the process can not change the configuration files for security reasons.

**Comment by WolfspiritM** [comment]
Thanks. You changed it for \*.key files but \*.pem files are also not readable by the user preventing the server to start aswell. As a workaround I changed the dh1024.pem to dh1024.key and it works now but I think it would be better to also set files with pem extension to the user, right?

**Comment by hfedcba** [comment]
Right, except for the CA key.

**Comment by WolfspiritM** [comment]
Thanks.
I'm sorry if I'm bugging you with this, but I've tested latest image.

Without `chown ${USER}:${USER} /etc/homegear/ca/private/*.pem` admin-ui shows a big fat red error:
`Warning: no CA found! Please click here to create one.`

Creating a CA will then fail.

**Comment by hfedcba** [comment]
Please bug me until it works ;-).

Homegear Management was started as user, that's why it didn't work. It has to run as root to fulfill it's purpose. It's made to do all the root operations. A new image is being built right now and should be online in the next minutes.

Cheers,

Sathya

**Comment by WolfspiritM** [comment]
@hfedcba Hi! I ran into this issue again after updating the image.
I think there is still a chmod missing for "*.crt" files. In my case after updating to a new image and creating a new container from it I always have to change start.sh to include
```
chown ${USER}:${USER} /etc/homegear/*.crt
```

I now even mounted a custom start.sh into the container to override /start.sh but I think this should be handled by the container itself.

**Comment by WolfspiritM** [comment]
@hfedcba Hi! I ran into this issue again now with the "homegear.crt" file. I think I've called it homegear.pem before which made it work but after creating new certificates I was sticking with the same naming scheme that homegear has as default which is homegear.crt but these files don't get the correct permissions on startup.

Adding
```
chown ${USER}:${USER} /etc/homegear/*.crt
```
to start.sh works.
